library/Booster/LLVM/Internal.hs

```diff
@@ -342,9 +347,11 @@ mkAPI dlib = flip runReaderT dlib $ do
                                         }
                                 len <- fromIntegral <$> peek lenPtr
                                 cstr <- peek strPtr
-                                BS.packCStringLen (cstr, len)
+                                result <- BS.packCStringLen (cstr, len)
+                                Foreign.free cstr
```

Comment:
Thanks for the spot here. I think I assumed that `packCStringLen` just wraps the pointer, but upon checking it copies the memory. I wonder if it would be worth passing this pointer tot he parse function and freeing after to avoid this copy? Not sure how big the returned data can be.... perhaps something to investigate later

Reply:
It might be worth it if we see performance issues related to this in a profile, but the advantage of this approach is that the native memory that is allocated is all isolated to this particular component, and all memory that gets used moving forward is just regular haskell heap memory. So I don't think we should prematurely optimize here.